### PR TITLE
Show generic error msg and hide stacktrace if feature flag is enabled

### PR DESCRIPTION
**Pull request by rahul-rahul-sp**
### CATEGORY

Choose one

- [ ] Bug Fix
- [X ] Enhancement (new features, refinement)
- [ ] Refactor
- [ ] Add tests
- [ ] Build / Development Environment
- [ ] Documentation

### SUMMARY

Show a generic error message and hide stacktrace if SHOW_STACKTRACE feature flag is disabled

### TEST PLAN

Tested by enabling and disabling SHOW_STACKTRACE feature flag.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by etr2460 — COMMENTED** [review]
I'm not a big fan of overloading the `SHOW_STACKTRACE` config variable like this, maybe it would make more sense to give this feature its own config option?

Taking a step back though, I'm curious the what the use case is here to overwrite every error message. Error messages generally should be specific to what actually went wrong so that the user can take necessary steps to correct the error. I acknowledge that a lot of the error messages in superset can be vague/technical (a lot are just passed on from the database backend) but I wonder if the "right way" to fix the issue is to improve our error messages in general, then use this one as a "last resort" type option. Thoughts?

**Comment by mistercrunch** [comment]
Showing stacktrace is generally super useful, but has potential security concerns. It may inform an attacker about specificities around the environment, and we don't control what gets bubbled up or not.

I think the intent with `SHOW_STACKTRACE` (config key was introduced super early by me) was to make showing the stacktrace optional. Ideally all areas of the code that may show a stacktrace should boil down to calling a same function so that logic isn't duplicated and can be centrally managed. For instance in the future we maybe want to redact only some parts of the stacktrace, or we may want to assign RBAC-style controls to it.

**Comment by etr2460** [comment]
I definitely agree that `SHOW_STACKTRACE` should remain and has value. My concern was with using it to replace all error messages with the same generic error. If we did want to do it, then we should use 2 different config vars to control that

**Comment by rahul-rahul-sp** [comment]
my intention to wrap the showing of the error message behind checking the feature flag was mostly to do with security concerns. by default the flag is enabled, so it shouldn't impact existing behavior. 
however if there is a different approach the community would like to take, i am all for it.

File: superset/views/base.py [hunk 1]
```diff
@@ -63,9 +63,12 @@ def get_error_msg():
 
 
 def json_error_response(msg=None, status=500, stacktrace=None, payload=None, link=None):
+    stack_enabled = conf.get('SHOW_STACKTRACE')
+    if not stack_enabled:
+        msg = 'An error has occurred.'
```
Comment (mistercrunch) [inline superset/views/base.py]: If a `msg` is passed we should just use it regardless of `SHOW_STACKTRACE`. `SHOW_STACKTRACE` should only affect the `stacktrace` key in the payload, not the msg itself.

**Review by mistercrunch — CHANGES_REQUESTED** [review]

**Review by mistercrunch — APPROVED** [review]

**Pull request merged**